app/modules/auth/api/auth.api.ts

```diff
@@ -0,0 +1,13 @@
+import { AxiosPromise } from "axios";
+import { User } from "../../common/models/User";
+
+export type LoginParams = Pick<User, 'login'>;
+
+export interface IAuthApi {
+    login(params: LoginParams): AxiosPromise<User>,
```

Comment:
Reusing `User` DTO for login purposes doesn't feel right, because it has implications on that DTO. E.g. it forces you to make `id` and `role` fields non-required, but in fact User DTO would always have them defined.